fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Scope.java

```diff
@@ -20,15 +23,51 @@ public class Scope {
     private final RelationFields relationFields;
     private final Map<String, CTERelation> cteQueries = Maps.newLinkedHashMap();
 
+    private List<PlaceHolderExpr> lambdaInputs = Lists.newArrayList();
+
+    private boolean isLambdaScope = false;
+
     public Scope(RelationId relationId, RelationFields relation) {
         this.relationId = relationId;
         this.relationFields = relation;
     }
 
+    public Scope(List<PlaceHolderExpr> exprs, Scope parent) {
```

Comment:
Why lambda function need an independent scope？

Reply:
due to nested lambda functions.